doc/api/util.md

```diff
@@ -330,6 +330,18 @@ changes:
 The `util.inspect()` method returns a string representation of `object` that is
 primarily useful for debugging. Additional `options` may be passed that alter
 certain aspects of the formatted string.
+`util.inspect()` will use the constructor's name and `@@toStringTag` to make an
+identifiable tag for an inspected value:
+
+```js
+class Foo {
+  get [Symbol.toStringTag]() {
+    return 'bar';
+  }
+}
+
+util.inspect(new Foo); // 'Foo [bar] {}'
+```
```

Comment:
I would also add some detail on what happens if `@@toStringTag` and construtor name are not present... e.g.

```
`util.inspect()` will use the constructor's name and `@@toStringTag` (if present) to make
an identifiable tag for an inspected value:
```

Then the examples:

```js
class Foo {
  get [Symbol.toStringTag]() {
    return 'bar';
  }
}

util.inspect(new Foo()); // 'Foo [bar] {}`
```

```js
function foo() {
  return new (class {
    get [Symbol.toStringTag]() { return 'bar'; }
  });
}
util.inspect(foo()); // '[bar] {}`
```

```js
class Foo {}

util.inspect(new Foo()); // 'Foo {}`
```

Reply:
that second one would actually be `Object [bar] {}`, i'm not sure that constructor.name will ever be absent.

Reply:
```js
> function m() {
... return new (class {})
... }
undefined
> m()
{}
> m().constructor.name
''
>
```

Reply:
```js
> new (class {})
{}
> new (class { get [Symbol.toStringTag]() { return 'tst'; } })
Object [tst] {}
```

not exactly sure whats going on here but it looks like if the class is empty the inheritance from `Object` is optimised away?